### ISSUE TITLE
No warning for indirect variable expansion when using POSIX sh

### ISSUE THREAD

**Issue by markgollnick** [koalaman/shellcheck#487]
###### Bug Description:

shellcheck produces no warning for [indirect variable expansion](https://www.gnu.org/software/bash/manual/bashref.html#Shell-Parameter-Expansion) when using the `#!/bin/sh` shebang, even though indirect variable expansion is functionality that is [not defined for POSIX sh](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02).

###### Bug Reproduction:

`test.sh`:

```bash
#!/bin/sh
export ULTIMATE="hello world"
export PENULTIMATE="ULTIMATE"
echo "${!PENULTIMATE}"
```

When run through bash in sh mode, the above script produces:

```
$ ./test.sh
hello world
```

**BUG:** When run through shellcheck, everything returns clean:

```
$ shellcheck test.sh
$ echo $?
0
```

**EXPECTATION:** When run through shellcheck, I would expect to receive an error (or at least a warning) due to the `#!/bin/sh` shebang, similar to the SC2039 warning that is given when a user attempts to use arrays despite specifying a shebang of `#!/bin/sh`, e.g.:

`test.sh`:

```bash
#!/bin/sh
ARGV=("one" "two" "three")
for x in "${ARGV[@]}"; do
  echo "$x"
done
```

```
$ ./test.sh
one
two
three
$ shellcheck test.sh

In test.sh line 2:
ARGV=("one" "two" "three")
     ^-- SC2039: In POSIX sh, arrays are not supported.


In test.sh line 3:
for x in "${ARGV[@]}"; do
          ^-- SC2039: In POSIX sh, array references are not supported.

$ echo $?
1
```

###### Environment/Versions:

- shellcheck v4.0.1

**Comment by markgollnick** [comment]
If I'm incorrect here and POSIX *does* define indirect variable expansion somewhere, maybe in one of their more recent docs/publications, feel free to close this… I just couldn't find anything online about it and assumed it was probably a bash feature, so even though it *works* (at least for bash in sh mode) I figured it might be worth adding a warning about it.

**Comment by koalaman** [comment]
You're absolutely right, well caught! This has been fixed in 1181c6b.
